homeassistant/components/prometheus/__init__.py

```diff
@@ -384,25 +456,89 @@ def state_as_number(state: State) -> float | None:
             value = None
         return value
 
-    @staticmethod
     def _labels(
+        self,
         state: State,
         extra_labels: dict[str, str] | None = None,
     ) -> dict[str, Any]:
         if extra_labels is None:
             extra_labels = {}
+
         labels = {
             "entity": state.entity_id,
             "domain": state.domain,
             "friendly_name": state.attributes.get(ATTR_FRIENDLY_NAME),
+            # Have to set the keys, because a metric cannot have variable labels
+            "area_id": "<no-area>",
+            "area_name": "<no-area>",
+            "area_norm_name": "<no-area>",
         }
+
+        labels.update(**self._make_area_labels(state.entity_id))
+
         if not labels.keys().isdisjoint(extra_labels.keys()):
             conflicting_keys = labels.keys() & extra_labels.keys()
             raise ValueError(
                 f"extra_labels contains conflicting keys: {conflicting_keys}"
             )
+
         return labels | extra_labels
 
+    def _make_area_labels(self, entity_id: str) -> dict[str, str]:
+        defaults = {
+            # Have to set the keys, because a metric cannot have variable labels
+            "area_id": "<no-area>",
+            "area_name": "<no-area>",
+            "area_norm_name": "<no-area>",
```

Comment:
Also let's cal this `area_normalized_name` to make it more explicit?

BTW, an alternative to adding all three labels to each metric might be to export an [info-style](https://www.robustperception.io/why-info-style-metrics-have-a-value-of-1/) metadata metric for areas, and join by area_id if area names are needed ([example](https://www.robustperception.io/exposing-the-software-version-to-prometheus/)). This might keep metric output more compact, and reduce metric churn when areas are renamed, but will require users who need these names to use more complex queries. I don't feel too strongly about one way or another.